ShoppingList/ShoppingList/ViewController.swift

```diff
@@ -30,42 +32,84 @@ class ViewController: UIViewController {
+            // 추가할 시 uncheckedList로 추가됨
             ShoppingManager.shared.append(name: name)
         } else if buttonType == .update && itemIndex != nil {
-            ShoppingManager.shared.update(name: name, index: itemIndex!)
+            // check uncheck 섹션 구분
+            // 해당 section 및 해당 row에 해당하는 데이터 수정
+            ShoppingManager.shared.update(name: name, section: itemSection!, index: itemIndex!)
             addButton.setTitle("저장", for: .normal)
         }
-        
         buttonType = .add
         mainTextField.text = nil
         mainTableView.reloadData()
     }
+    
+    @IBAction func checkButtonClicked(_ sender: CheckButton) {
+        guard let row = sender.row else { return }
+        guard let section = sender.section else { return }
+        guard let shoppingListItem = ShoppingManager.shared.getShoppingItem(section: section, index: row) else { return }
+        
+        shoppingListItem.check.toggle()
+        ShoppingManager.shared.update(isChecked: shoppingListItem.check, section: section, index: row)
+        mainTableView.reloadData()
+    }
+    
+    @IBAction func sortButtonClicked(_ sender: UIBarButtonItem) {
+        let alert = UIAlertController(title: "정렬 방식", message: "", preferredStyle: .actionSheet)
+        let cancel = UIAlertAction(title: "취소", style: .cancel)
+        let descending = UIAlertAction(title: "내림차순", style: .default, handler: { _ in
+            ShoppingManager.shared.highToLow()
+            self.mainTableView.reloadData()
+        })
+        let ascending = UIAlertAction(title: "오름차순", style: .default, handler: { _ in
+            ShoppingManager.shared.lowToHigh()
+            self.mainTableView.reloadData()
+        })
+        alert.addAction(cancel)
+        alert.addAction(descending)
+        alert.addAction(ascending)
+        self.present(alert, animated: true)
+    }
+    
 }
 
 extension ViewController: UITableViewDelegate, UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        ShoppingManager.shared.sectionList.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        ShoppingManager.shared.sectionList[section]
+    }
+    
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return ShoppingManager.shared.count()
+        return ShoppingManager.shared.count(section: section)
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         guard let cell = self.mainTableView.dequeueReusableCell(withIdentifier: "customCell", for: indexPath) as? CustomTableViewCell else { return UITableViewCell() }
         
-        let item = ShoppingManager.shared.getShoppingItem(index: indexPath.row)
-        cell.mainLabel.text = item.name
+        let item = ShoppingManager.shared.getShoppingItem(section: indexPath.section, index: indexPath.row)
```

Comment:
section과 row 정보 둘다 인자로 받고자 한다면, 이름을 좀더 명시적으로 아래와 같이 바꿔주는게 좋습니다.
index라는 단어는 section과 row를 모두 포함하고 있는 단어입니다!
```suggestion
        let item = ShoppingManager.shared.getShoppingItem(section: indexPath.section, row: indexPath.row)
```

Reply:
@keenkim1202 수정 완료했습니다!